shpc/main/wrappers/__init__.py

```diff
@@ -22,40 +22,54 @@ def generate(image, container, config, **kwargs):
     settings = container.settings
     aliases = kwargs.get("aliases", [])
 
-    # Global wrapper scripts are generated via aliases
-    for container_tech in ["docker", "podman", "singularity"]:
-        template_name = settings.wrapper_scripts.get(container_tech)
-        if template_name and aliases and container_tech == container.command:
-            wrapper = get_wrapper_script(template_name)(
-                settings=settings,
-                wrapper_template=template_name,
-                config=config,
-                container=container,
-                image=image,
-                **kwargs
-            )
-            generated += wrapper.generate_aliases()
+    constructor_kwargs = {
+        "settings": settings,
+        "config": config,
+        "container": container,
+        "image": image,
+        **kwargs,
+    }
+
+    # Default wrapper for this container technology. Will be used for all
+    # aliases (unless overriden)
+    default_wrapper = None
+    default_template_name = settings.wrapper_scripts.get(container.command)
+    if default_template_name:
+        default_wrapper = WrapperScript(
+            wrapper_template=default_template_name, **constructor_kwargs
+        )
+        # No extra parameters -> only look in the global locations
+        default_wrapper.load_template()
 
     # Container level wrapper scripts (allow eventually supporting custom podman)
     scripts = {
         "singularity": config.singularity_scripts,
         "docker": config.docker_scripts,
         "podman": config.docker_scripts,
     }
-    for command, listing in scripts.items():
-
-        # Don't look if no custom container.yaml scripts OR wrong container tech
-        if not listing or container.templatefile != command:
-            continue
-        for alias, template_name in listing.items():
-            wrapper = get_wrapper_script(template_name)(
-                settings=settings,
-                wrapper_template=template_name,
-                config=config,
-                container=container,
-                image=image,
-                **kwargs
-            )
-            generated += wrapper.generate(alias)
+    # Additional commands defined in the custom container.yaml script section
+    container_templates = scripts.get(container.templatefile, [])
+    container_wrappers = {}
+    for alias, template in container_templates.items():
```

Comment:
Ah I think I see your confusion here - we have the idea of an "alias" as in "the name of the command we are writing a wrapper for instead (e.g., fork)" and then the aliases proper, which the module templates use that read in the aliases (and map to a common structure) to render into the template.

Reply:
It might be helpful to update the docs in places where I didn't do a good job of explaining what to expect.